backend/src/main.rs

```diff
@@ -0,0 +1,33 @@
+#![feature(plugin)]
+#![plugin(rocket_codegen)]
+
+extern crate rocket;
+
+extern crate newsletter;
```

Comment:
I'm not sure to understand why there is two crates here. I think it's a mis-understanding of how Rust works.

Basically, a crate is a project. It's composed of multiple files.
A module is a file containing items like functions, types, global variables etc. It can be made of other modules.

The main module for a binary project is named `main.rs`, and the main module for a library project is named `lib.rs`.

In this project, you have both. And therefore, this leads you with two crates. Named the same.

That's probably not what you wanted, instead you probably wanted an other module.
You can look at the corresponding section in the Rust book and take a look at the code in `nest` or `nest-server` to see how things should be organised.